src/lib.rs

```diff
@@ -57,24 +40,33 @@ impl UPTransportZenoh {
     ///
     /// * `config` - The Zenoh configuration to use.
     ///   Please refer to the [Zenoh documentation](https://zenoh.io/docs/manual/configuration/) for details.
-    /// * `uri` - Local `UUri`. Note that the Authority of the `UUri` MUST be non-empty and the resource ID should be non-zero.
+    /// * `uri` - The uProtocol URI identifying the local uEntity. Its authority MUST be non-empty and the
+    ///   resource ID must be 0.
+    /// * `max_listeners` - The maximum number of listeners that can be registered with this transport.
     ///
     /// # Errors
-    /// Will return `Err` if unable to create `UPTransportZenoh`
+    ///
+    /// Returns an error if the transport cannot be created. Returns an error with `UCode::ILLEGAL_ARGUMENT`,
+    /// if the given URI has an empty authority or a non-zero resource ID.
     ///
     /// # Examples
     ///
     /// ```
     /// #[tokio::main]
     /// # async fn main() {
     /// use up_transport_zenoh::{zenoh_config, UPTransportZenoh};
-    /// let uptransport =
-    ///     UPTransportZenoh::new(zenoh_config::Config::default(), "//MyAuthName/ABCD/1/0")
+    ///
+    /// assert!(
+    ///     UPTransportZenoh::new(zenoh_config::Config::default(), "//MyAuthName/ABCD/1/0", 10)
     ///         .await
-    ///         .unwrap();
+    ///         .is_ok());
     /// # }
     /// ```
-    pub async fn new<U>(config: zenoh_config::Config, uri: U) -> Result<UPTransportZenoh, UStatus>
+    pub async fn new<U>(
```

Comment:
I'm thinking that we can have another API (`pub async fn new_with_listeners_num`) and use a default value here to avoid the API changes.
In fact, a better way is to create a UPTransportZenohBuilder to accept the arguments. This will provide the flexibility to add arguments in the future.

Reply:
I guess a builder would be nice, indeed. FMPOV we should not be too concerned about breaking the API at this early stage. I'll update the PR and add a builder ...